XenModel/Actions/Updates/DownloadAndUnzipXenServerPatchAction.cs

```diff
@@ -123,14 +123,22 @@ private void ExtractFile()
 
                     while (iterator.HasNext())
                     {
-                        if (Path.GetExtension(iterator.CurrentFileName()) == "." + updateFileExtension)
+                        string currentExtension = Path.GetExtension(iterator.CurrentFileName());
+
+                        if (!string.IsNullOrEmpty(updateFileExtension) && currentExtension == "." + updateFileExtension
+                            || currentExtension == ".iso" || currentExtension == ".xsupdate")

```

Comment:
We cannot use hardcode "xsupdate" here, we should use Branding.Update instead.
Also, I think it would be better to put ".iso" in a constant.